src/pages/Dashboard/widgets/BatchTransactions/BatchTransactions.tsx

```diff
@@ -1,137 +1,35 @@
-import { useEffect, useState } from 'react';
 import {
   faPaperPlane,
   faArrowsRotate
 } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { useGetBatches } from '@multiversx/sdk-dapp/hooks/transactions/batch/useGetBatches';
 import { Button } from 'components/Button';
-import {
-  OutputContainer,
-  TransactionsOutput
-} from 'components/OutputContainer';
-import {
-  useGetAccountInfo,
-  useGetNetworkConfig,
-  useGetPendingTransactions
-} from 'hooks';
-import { SessionEnum } from 'localConstants/session';
-import { SignedTransactionType, WidgetProps } from 'types';
-import { useBatchTransactionContext } from 'wrappers';
-import { useSendSignedTransactions } from './hooks';
-import {
-  sendBatchTransactions,
-  signAndAutoSendBatchTransactions,
-  swapAndLockTokens
-} from './helpers';
-
-export const BatchTransactions = ({ callbackRoute }: WidgetProps) => {
-  const { setSendBatchTransactionsOnDemand } = useBatchTransactionContext();
-  const { address, account } = useGetAccountInfo();
-  const { network } = useGetNetworkConfig();
-  const { batches } = useGetBatches();
-  const { hasPendingTransactions } = useGetPendingTransactions();
-  const [trackBatchId, setTrackBatchId] = useState(
-    sessionStorage.getItem(SessionEnum.batchId)
-  );
-
-  const [stateTransactions, setStateTransactions] = useState<
-    SignedTransactionType[] | null
-  >(null);
-  const [currentSessionId, setCurrentSessionId] = useState(
-    sessionStorage.getItem(SessionEnum.signedSessionId) || ''
-  );
-
-  const { batchId, setBatchSessionId } = useSendSignedTransactions({
-    signedSessionId: currentSessionId
-  });
-
-  // If manual batch transactions are executed, track the batchId
-  useEffect(() => {
-    if (batchId) {
-      setTrackBatchId(batchId);
-    }
-  }, [batchId]);
-
-  useEffect(() => {
-    if (trackBatchId && batches[trackBatchId]) {
-      setStateTransactions(batches[trackBatchId].transactions.flat());
-    }
-  }, [trackBatchId, batches]);
-
-  const executeSignAndAutoSendBatchTransactions = async () => {
-    setSendBatchTransactionsOnDemand(false);
-
-    const { batchId } = await signAndAutoSendBatchTransactions({
-      address,
-      nonce: account.nonce,
-      chainID: network.chainId,
-      callbackRoute
-    });
+import { TransactionsOutput } from 'components/OutputContainer/components';
+import { OutputContainer } from 'components/OutputContainer/OutputContainer';
+import { useSendBatchTransaction } from 'hooks';
```

Comment:
why create a hook and not make the transactions here to be more like initial template?